api/server.py

```diff
@@ -281,34 +280,42 @@ def groupby_element(event_type, element):
     Returns:
         JSON/Dict top ten element in event type
     """
-    abort(404) if (event_type not in event_types or element not in group_by_elements) else True
-
+    abort(404) if (
+            event_type not in event_types or element not in group_by_elements
+    ) else True
     date = get_value_from_request("date")
     filter_by = get_value_from_request('filter_by')
     element_value = get_value_from_request(filter_by)
+    conditions = [condition for condition in [
+        filter_by_date(date)['query'],
+        filter_by_element(filter_by, element_value)['query']
+    ] if condition]
```

Comment:
why do you need to check the last "if" condition? is it ever going to be None?

Reply:
```py
def filter_by_element(filter_by, element_value):
    return {
        "query": {
            "multi_match": {
                "query": element_value,
                "fields": ['*']
            }
        }
    } if filter_by and element_value else {
        "query": {}
    }
```

if the `if` condition fails in this case it returns an empty query and same for `filter_by_date` function which is treated as error by elasticsearch. So I added the If condition to remove empty dictionaries